src/error.rs

```diff
@@ -1,3 +1,28 @@
 use std::{error, result};
+use std::fmt::Display;
 
 pub type Result<T> = result::Result<T, Box<error::Error>>;
+
+pub trait OrExit {
```

Comment:
Introducing a new trait is pretty common way to extend behaviour of some object. This pattern is called 'extension trait'. For example, [tokio-openssl](https://docs.rs/tokio-openssl/0.1.3/tokio_openssl/) uses this pattern to add asyncronous versions of connect() and accept() to corresponding types from openssl crate.